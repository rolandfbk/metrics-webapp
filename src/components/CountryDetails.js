import { React, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';
import { getDataFromApi } from '../redux/metric-redux';

const CountryDetails = () => {
  const listing = useSelector((state) => state.metricsReducer);
  const { id } = useParams();
  console.log(typeof(id));

  const dispatch = useDispatch();

  const details = listing.filter((country) => { return country.id === Number(id) });
  console.log(details);

  useEffect(() => {
    if (listing.length === 0) {
      getDataFromApi(dispatch);
    }
  }, [dispatch]);

  return (
    <div className="rocket">
      {details.map((crt) => {
        return (
          <table>
            <tr key={1}>
              <td className="my-profil-mission">{crt.shortName}</td>
            </tr>
            <tr key={2}>
              <td className="my-profil-mission">{crt.fullName}</td>
            </tr>
            <tr key={3}>
              <td className="my-profil-mission">{crt.currencyName}</td>
            </tr>
            <tr key={4}>
              <td className="my-profil-mission">{crt.currencySymbol}</td>
            </tr>
            <tr key={5}>
              <td className="my-profil-mission">{crt.capital}</td>
            </tr>
            <tr key={6}>
              <td className="my-profil-mission">{crt.subregion}</td>
            </tr>
            <tr key={7}>
              <td className="my-profil-mission">{crt.languages}</td>
            </tr>
            <tr key={8}>
              <td className="my-profil-mission">{crt.population}</td>
            </tr>
            <tr key={9}>
              <td className="my-profil-mission">{crt.fifa}</td>
            </tr>
            <tr key={10}>
              <td className="my-profil-mission">{crt.carSigns}</td>
            </tr>
            <tr key={11}>
              <td className="my-profil-mission">{crt.carSide}</td>
            </tr>
            <tr key={12}>
              <td className="my-profil-mission">{crt.timezones}</td>
            </tr>
            <tr key={13}>
              <td className="my-profil-mission">{crt.continents}</td>
            </tr>
            <tr key={14}>
              <td className="my-profil-mission">{crt.flags}</td>
            </tr>
            <tr key={15}>
              <td className="my-profil-mission">{crt.coatOfArms}</td>
            </tr>
            <tr key={16}>
              <td className="my-profil-mission">{crt.startOfWeek}</td>
            </tr>
          </table>
        );
      })}
    </div>
  );
};

export default CountryDetails;