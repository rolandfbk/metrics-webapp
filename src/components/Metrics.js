import { React, useEffect } from 'react';
import { useDispatch, useSelector, shallowEqual } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowAltCircleRight } from '@fortawesome/free-solid-svg-icons';
import { getDataFromApi } from '../redux/metric-redux';

const Metrics = () => {
  const listing = useSelector((state) => state.metricsReducer, shallowEqual);

  const dispatch = useDispatch();

  useEffect(() => {
    if (listing.length === 0) {
      getDataFromApi(dispatch);
    }
  }, [dispatch]);

  return (
    <div className="country">
      <div className="countryHeading">
        <div className="countryHeadingImage">
          <img src="africa.png" alt="Africa map" />
        </div>
        <div className="countryHeadingTitle">
          <div>
            <h1>AFRICA</h1>
            <span>The heart of hope</span>
          </div>
        </div>
      </div>
      <div className="countrylistTitle">
        <h3>LIST OF COUNTRIES</h3>
      </div>
      <ul>
        {listing.map((country) => (
          <li className="countrylistItem" style={{ backgroundImage: `linear-gradient(rgb(66 105 176 / 80%), rgb(66 105 176 / 80%)), url(${country.flags})` }} key={country.id}>
            <NavLink className="active-link" to={`/countryDetails/${country.id}`}>
              <div>
                <FontAwesomeIcon icon={faArrowAltCircleRight} />
              </div>
              <div>
                <h2>{country.shortName}</h2>
                <span>{country.capital}</span>
              </div>
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Metrics;
