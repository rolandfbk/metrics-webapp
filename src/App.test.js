import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Metrics from './components/Metrics';
import CountryDetails from './components/CountryDetails';
import store from './redux/ConfigureStore';

describe('component test', () => {
  const MetricsProvider = () => (
    <Provider store={store}>
      <Metrics />
    </Provider>
  );

  const CountryDetailsProvider = () => (
    <Provider store={store}>
      <CountryDetails />
    </Provider>
  );

  it('Metrics renders correctly', () => {
    const tree = renderer.create(<MetricsProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Simulation of user click on Morocco', () => {
    const tree = renderer.create(<CountryDetailsProvider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
