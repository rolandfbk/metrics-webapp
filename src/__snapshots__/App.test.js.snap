// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test Metrics renders correctly 1`] = `
<div
  className="country"
>
  <div
    className="countryHeading"
  >
    <div
      className="countryHeadingImage"
    >
      <img
        alt="Africa map"
        src="africa.png"
      />
    </div>
    <div
      className="countryHeadingTitle"
    >
      <div>
        <h1>
          AFRICA
        </h1>
        <span>
          The heart of hope
        </span>
      </div>
    </div>
  </div>
  <div
    className="countrylistTitle"
  >
    <h3>
      LIST OF COUNTRIES
    </h3>
  </div>
  <ul />
</div>
`;

exports[`component test Simulation of user click on Morocco 1`] = `
<div
  className="country"
/>
`;

exports[`component test renders correctly 1`] = `
<div
  className="country"
>
  <div
    className="countryHeading"
  >
    <div
      className="countryHeadingImage"
    >
      <img
        alt="Africa map"
        src="africa.png"
      />
    </div>
    <div
      className="countryHeadingTitle"
    >
      <div>
        <h1>
          AFRICA
        </h1>
        <span>
          The heart of hope
        </span>
      </div>
    </div>
  </div>
  <div
    className="countrylistTitle"
  >
    <h3>
      LIST OF COUNTRIES
    </h3>
  </div>
  <ul />
</div>
`;
